app.py

```diff
@@ -15,40 +15,60 @@ def index():
+        apiOutputs = []
 
-        # get lat/long of airports
-        NotamRequest.startLat, NotamRequest.startLong = alc.get_lat_and_lon(NotamRequest.startAirport)
-        NotamRequest.destLat, NotamRequest.destLong = alc.get_lat_and_lon(NotamRequest.destAirport)
-        
-        NotamRequest.radius = int(NotamRequest.radius)
-        NotamRequest.pathWidth = int(NotamRequest.pathWidth)
-
-        # get the list of coordinates that need to be called to cover area
-        coordList = MinimalCirclesPath.getPath(NotamRequest.startLat, 
-                                               NotamRequest.startLong,
-                                               NotamRequest.destLat,
-                                               NotamRequest.destLong, 
-                                               NotamRequest.radius, # circle radius
-                                               NotamRequest.pathWidth) # path width
+        #i is used to track where we are in the array of airports
+        i = 0
         # start timer
         startTime = time.time() 
+        
+        #len(airports) -1 ensures the loop treats i as the starting location for each iteration and prevents out of index errors
+        while i < len(airports) - 1:
+            
+            # get lat/long of airports
+            NotamRequest.startLat, NotamRequest.startLong = alc.get_lat_and_lon(airports[i])
+            NotamRequest.destLat, NotamRequest.destLong = alc.get_lat_and_lon(airports[i+1])
+        
+            NotamRequest.radius = int(NotamRequest.radius)
+            NotamRequest.pathWidth = int(NotamRequest.pathWidth)
 
-        # call the API for each point
-        print("LOADING...")
 
-        # apiOutputs = [ GetNOTAM.getNotam( NotamRequest.effectiveStartDate,
-        #                                     NotamRequest.effectiveEndDate,
-        #                                     longitude, # longitude
-        #                                     latitude, # latitude
-        #                                     1, # page num
-        #                                     NotamRequest.radius) #page num here is one temporarily
-        #                                     for latitude, longitude in coordList ]
-        apiOutputs = []
+            #handles every other path thus prevents double calls for the same location being the final location for one path and the start location for the next path
+            if i >= 1:
+                # updates the start to a be outside of the previous call area from the start being the previous destination
+                bearing = MinimalCirclesPath.calculateBearing(NotamRequest.startLat, NotamRequest.startLong, NotamRequest.destLat, NotamRequest.destLong)
+                # updatedstart is the lat and long outside of the previous call from being the final location
+                updatedStart = MinimalCirclesPath.nextPoint(NotamRequest.startLat, NotamRequest.startLong, bearing, NotamRequest.radius)

```

Comment:
the fourth variable of nextPoint is not radius. It is the stepDistance so this is not getting the correct updated point

Reply:
a fix for this and a way to minimize this big if/else statement is that if we are not getting the first segment we can just ignore the first point. This way you do not have to manually update the bearing and start from in here.